### DISCUSSION TITLE
Python usage of option "threads"

### DISCUSSION THREAD

**Discussion by skartekko** [m-bain/whisperX#1251] [Q&A]
Hi there, how can I use the option "--threads" with Python?
Many thanks in advance!

**Comment by skartekko** [comment]
@m-bain : Do you have any advice, how to use the options 
`--threads`
and 
`--beam_size`
in my python script with WhisperX?
Thank you very much!

**Comment by sorgfresser** [comment]
Regarding threads: `torch.set_num_threads(threads)`
And beam size: pass it to load_model as part of the dict used with `asr_options` keyword.
So simply do `load_model(..., asr_options={'beam_size': 5})` for a beam size of 5

**Comment by skartekko** [comment]
> orch.set_num_threads(threads

Thank you @sorgfresser !

beam size = 5 is recommended for accurate transcription, right?

**Comment by sorgfresser** [comment]
At least it's the default. You could experience with it, but beam size 5 seems to work very well.

**Comment by papatistos** [comment]
@sorgfresser when I use `torch.set_num_threads(6)`I get 

````
[W ParallelNative.cpp:230] Warning: Cannot set number of intraop threads after parallel work has started or after set_num_threads call when using native parallel backend (function set_num_threads)
````

How/where exactly do I set the number of threads?
